src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,154 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_SIZE = 10;
+    private static final int NOT_FOUND_INDEX = -1;
+    private T[] content;
+    private int size;
+
+    public ArrayList() {
+        content = (T[]) new Object[DEFAULT_SIZE];
+    }
+
+    public ArrayList(int customSize) {
+        content = (T[]) new Object[customSize];
+    }
+
     @Override
     public void add(T value) {
-
+        growIfNeeded();
+        content[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (!(index >= 0 && index <= size)) {
+            String errorMessage = String.format("Can't add value at index %s. It's out of bounds.",
+                    index);
+            throw new ArrayListIndexOutOfBoundsException(errorMessage);
+        }
+        growIfNeeded();
+        System.arraycopy(content, index, content, index + 1, size++ - index);
```

Comment:
Let's separate size growing and array copying, looks not so obvious now.

```suggestion
        System.arraycopy(content, index, content, index + 1, size - index);
        size++;
```